sys/vm/vm_map.c

```diff
@@ -4433,6 +4433,7 @@ vm_map_delete(vm_map_t map, vm_offset_t start, vm_offset_t end,
 			if ((entry->eflags & MAP_ENTRY_UNMAPPED) == 0) {
 				vm_map_entry_clean(map, entry);
 				/* XXX-AM: How do we reset maxprot? */
+				vm_object_deallocate(entry->object.vm_object);
```

Comment:
Hmm, I wonder if this needs a check on `MAP_ENTRY_IS_SUB_MAP` to match what is done in `vm_map_entry_deallocate`?

Reply:
Yes, that makes sense to me. I've pushed a commit with the check. Do we need to do some other clean up if it is a submap? I don't know how submaps are used.

For anyone following this PR, this is the first time I work with the VM subsystem.

Reply:
I think for documentation purposes (i.e. reading the code) it's best to keep the check.

Reply:
Ok, makes sense.